File: editor-api/src/main/groovy/ish/oncourse/willow/editor/service/RedirectApi.java
package ish.oncourse.willow.editor.service;

import ish.oncourse.willow.editor.model.common.CommonError;
import ish.oncourse.willow.editor.model.redirect.RedirectSettings;

import java.io.InputStream;
import java.io.OutputStream;
import java.util.List;
import java.util.Map;
import javax.ws.rs.*;
import javax.ws.rs.core.Response;
import javax.ws.rs.core.MediaType;

@Path("/")
public interface RedirectApi  {

    @GET
    @Path("/redirect.list")
    @Produces({ "application/json" })
    @AuthFilter
    RedirectSettings redirectListGet();

    @POST
    @Path("/redirect.update")
    @Consumes({ "application/json" })
    @Produces({ "application/json" })
    @AuthFilter
    RedirectSettings redirectUpdatePost(RedirectSettings redirectSettingsRequest);
}

File: editor/src/js/http/RedirectApi.ts
import {HttpService} from "../common/services/HttpService";
import {CommonError} from "../model/common/CommonError";
import {RedirectSettings} from "../model/redirect/RedirectSettings";

export class RedirectApi {
  constructor(private http: HttpService) {
  }

  redirectListGet(): Promise<RedirectSettings> {
    return this.http.GET(`/redirect.list`);
  }
  redirectUpdatePost(redirectSettingsRequest: RedirectSettings): Promise<RedirectSettings> {
    return this.http.POST(`/redirect.update`, redirectSettingsRequest);
  }
}
